pysat/utils/io.py

```diff
@@ -782,10 +1414,90 @@ def load_netcdf_xarray(fnames, strict_meta=False, file_format='NETCDF4',
     return data, meta
 
 
-def inst_to_netcdf(inst, fname, base_instrument=None, epoch_name='Epoch',
+def return_epoch_metadata(inst, epoch_name):
+    """Return epoch or time-index metadata.
+
+    Parameters
+    ----------
+    inst : pysat.Instrument
+        Instrument object with data and metadata.
+    epoch_name : str
+        Data key for time-index or epoch data
+
+    Returns
+    -------
+    meta_dict : dict
+        Dictionary with epoch metadata, keyed by
+        metadata label.
+
+    """
+    # Get existing meta data
+    if epoch_name in inst.meta:
+        new_dict = inst.meta[inst.meta.var_case_name(epoch_name)].to_dict()
+    else:
+        new_dict = {}
+
+    # Update basic labels, if they are missing
+    epoch_label = 'Milliseconds since 1970-1-1 00:00:00'
+    basic_labels = [inst.meta.labels.units]
+
+    for label in basic_labels:
+        if label not in new_dict or len(new_dict[label]) == 0:
+            new_dict[label] = epoch_label
+
+    # Assign name
+    new_dict[inst.meta.labels.name] = epoch_name
+
+    # Update the time standards
+    time_dict = {'calendar': 'standard', 'Format': 'i8', 'Var_Type': 'data',
+                 'Time_Base': epoch_label, 'Time_Scale': 'UTC'}
+
+    if inst.index.is_monotonic_increasing:
+        time_dict['MonoTon'] = 'increase'
+    elif inst.index.is_monotonic_decreasing:
+        time_dict['MonoTon'] = 'decrease'
+
+    new_dict.update(time_dict)
+
+    return new_dict
+
+
+def xarray_vars_no_time(data, time_label='time'):
+    """Return all DataSet variables except `time_label` dimension.
+
+    If `time_label` not found, removes first 1D dimension with datetime data.
+
+    Parameters
+    ----------
+    data : xarray.Dataset
+
+    Returns
+    -------
+    variables :
```

Comment:
Missing type(s).  Also doesn't match return in all cases.

Docstring is also missing Raises section.